internal/api/fetcher/fetcher.go

```diff
@@ -46,6 +48,28 @@ type apiObject interface {
 	URL() (string, error)
 }
 
+// FetchFromURL fetches the content from the given URL and returns the content as a byte slice.
+func FetchFromURL(ctx context.Context, client HTTPClient, sourceURL *url.URL) ([]byte, error) {
```

Comment:
Please keep the function as `getFromURL` within the local scope of the pkg that needs it. If needed, copy it.

`fetcher` should only be used by implementing the `apiObject` interface and supplying the new type to `fetch`.
`FetchFromURL` is only necessary since we are not handling measurements according to that scheme, yet. I am working towards changing that right now.